sopel/trigger.py

```diff
@@ -163,21 +178,25 @@ def __init__(self, own_nick, line, url_schemes=None):
 
         self.event = self.args[0]
         self.args = self.args[1:]
-        components = PreTrigger.component_regex.match(self.hostmask or '').groups()
-        self.nick, self.user, self.host = components
-        self.nick = tools.Identifier(self.nick)
+
+        # The regex will always match any string, even an empty one
+        components_match = cast(
+            Match, PreTrigger.component_regex.match(self.hostmask or ''))
+        nick, self.user, self.host = components_match.groups()
+        self.nick: identifiers.Identifier = self.make_identifier(nick)
 
         # If we have arguments, the first one is the sender
         # Unless it's a QUIT event
+        target: Optional[identifiers.Identifier] = None
+
         if self.args and self.event != 'QUIT':
-            target = tools.Identifier(self.args[0])
-        else:
-            target = None
+            target = self.make_identifier(self.args[0])
+
+            # Unless we're messaging the bot directly, in which case that
+            # second arg will be our bot's name.
+            if target.lower() == own_nick.lower():
+                target = self.nick
 
-        # Unless we're messaging the bot directly, in which case that second
-        # arg will be our bot's name.
-        if target and target.lower() == own_nick.lower():
-            target = self.nick
         self.sender = target
```

Comment:
I've been staring at this for 5 minutes (not counting however long I stared at it in the last review) and… Well, it'll be faster just to ask what you're trying to accomplish with the restructuring. 😅

Reply:
The core idea is that it doesn't make sense to have a second `if target` if we can put that directly within the first `if` that ensure it's not `None`:

1. The target is None by default, with a type hint for an optional Identifier.
2. If it's not a `QUIT` event, it becomes an Identifier (before: it would either become an Identifier or be `None`)
3. If we just created the `target`, we can check if it's the bot's nick or not

The diff is not the most readable things to be honest.